ui/pages/confirmations/components/confirm/row/dataTree.tsx

```diff
@@ -81,13 +81,18 @@ const DataField = memo(
       );
     }
     if (isPermit && label === 'value') {
-      const valueBN = new BigNumber(
-        parseInt(value, 10) / Math.pow(10, tokenDecimals),
-      );
-      const tokenValue = formatAmount('en-US', valueBN);
+      const valueBN = new BigNumber(value);
+      const diviserBN = new BigNumber(10).pow(tokenDecimals);
+      const resultBn = valueBN.div(diviserBN);
+
+      // FIXME - Precision may be lost for large values when using formatAmount
+      /** @see {@link https://github.com/MetaMask/metamask-extension/issues/25755} */
+      const tokenValue = formatAmount('en-US', resultBn);
```

Comment:
As this logic has been extended further and is duplicated, could we create a util such as `formatStringAmountWithDecimals`?

Reply:
Agreed and similar to the comment in https://github.com/MetaMask/metamask-extension/pull/25741#discussion_r1673863513, we could tackle this when we address the new upcoming number displays as this is out of scope for the fix